Add %current_month% macro

arnaudbroes opened the pull request:
Issue #2301.

## Proposed changes

Adds the %current_month% title format macro to add the current month (localized) to the title.

## Types of changes

What types of changes does your code introduce?

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I've selected the appropriate branch (ie x.y rather than master).
- [x] I've created an appropriate title, descriptive of what the PR does.
- [x] Travis-ci runs with no errors.

## Testing instructions

Check this in English and an non-Latin alphabet language such as Japanese (to see if it is localized) for the following title format fields:

- Homepage
- Post
- Page
- Category
- Description

You can ignore all (other) Grunt errors. The ones in these two files have been fixed but I think new Grunt errors are showing up now because @EkoJR changed some of the rules.

## Further comments

I'm not adding a unit test because #2387 needs to be resolved first.

wpsmort commented:
This is tested and is good.

michaeltorbert commented:
On second thought. Since we're localizing this, I can see a need for someone who has a site's backend in their native language for ease of use, but wants the front-end in English for wider reach.
Maybe we should have something like %current_month% and %current_month_i18n%?

arnaudbroes commented:
Moving this back to development to add a second macro.

arnaudbroes commented:
@wpsmort I've added the second %current_month_i18n% macro and changed %current_month% macro to the unlocalized name of the current month. Please test both.

wpsmort commented:
This is tested and is good.
